### ISSUE TITLE
stdin as Input

### ISSUE THREAD

**Issue by arrmo** [Microsoft/PTVS#1246]
Hi,

Sorry if this is a dumb question, but I can't seem to get it working .. :(.

Trying to use stdin, to debug a Python file that will be used on Windows and Linux. Inside Visual Studio, I have set the Script Parameters as < inputfiile ... but it's not showing up as stdin.

Has anyone gotten this working?

Thanks!

**Comment by zooba** [comment]
That _might_ work if you ensure that the process will pause at the end, but only because we make that work by passing it to `cmd.exe`. The `< file` syntax is part of `cmd.exe` and not universal to Windows, so really we'd need to emulate it to enable it.

Most scripts that support this also allow you to pass a filename as an argument, so I'd suggest adding support for that. Then you can just pass the filename directly while testing.

**Comment by arrmo** [comment]
Yep, I have passed the file by name ... but I really want to pipe the input - so the same code works both on Windows, with a file, and Linux with piped input (where a file does not exist, it's streamed input). So only common approach is piped / redirected input.

**Comment by zooba** [comment]
To make this work we'll need to add a new option for providing a file to stream to stdin. Until then, you're best to test manually from the command line (you can right-click your project and choose "Open command prompt here" to open one in the right place and where `python.exe` will work).

**Comment by arrmo** [comment]
I'll do that - thanks!
